addons/mail/models/mail_message.py

```diff
@@ -1208,10 +1208,39 @@ def _bus_notification_target(self):
     def _cleanup_side_records(self):
         """ Clean related data: notifications, stars, ... to avoid lingering
         notifications / unreachable counters with void messages notably. """
+        outdated_starred_partners = self.starred_partner_ids
         self.write({
             'starred_partner_ids': [(5, 0, 0)],
             'notification_ids': [(5, 0, 0)],
         })
+        if not outdated_starred_partners:
+            return
```

Comment:
This is a rare case where I would avoid the quick return, in case we need to add something else at the end of the method we don't have to re-indent everything and change the condition (the method being about cleaning generally speaking and not about starred).